src/libraries/System.Private.CoreLib/src/System/Type.cs

```diff
@@ -78,7 +79,7 @@ public virtual Type[] GetGenericParameterConstraints()
         public bool IsSealed => (GetAttributeFlagsImpl() & TypeAttributes.Sealed) != 0;
         public bool IsSpecialName => (GetAttributeFlagsImpl() & TypeAttributes.SpecialName) != 0;
 
-        public bool IsClass => (GetAttributeFlagsImpl() & TypeAttributes.ClassSemanticsMask) == TypeAttributes.Class && !IsValueType;
+        public bool IsClass { [Intrinsic] get => (GetAttributeFlagsImpl() & TypeAttributes.ClassSemanticsMask) == TypeAttributes.Class && !IsValueType; }
```

Comment:
Are these the highest value methods that it is worth it to handle as intrinsics?

I agree that `IsValueType` makes sense to handle as intrinsics. It can be used to replace the `default != null` check that is used today. I am not sure about real use cases for IsPrimitive or IsClass.

The one method that would be nice to handle as intrinsic is `IsAssignableFrom`.

Reply:
@jkotas I can leave only `IsValueType` it should simplify the PR as I also can't find `IsClass` and `IsPrimitive` usages in projects I have locally. Will leave `IsAssignableFrom` (and `IsSubclassOf`?) for the next PR.